### PR TITLE
[QNEBE-666] Fix for cve-2007-4559

### PR DESCRIPTION
**Pull request by QFer**
* Fix from https://github.com/QuTech-Delft/qne-adk/pull/51 adjusted to pass the pipe lines
* Added unit tests for the changes
* Completed unit tests for resource_manager

### CODE REVIEW

File: src/adk/managers/resource_manager.py [hunk 1]
```diff
@@ -77,7 +78,26 @@ def generate_resources(self, qne_client: QneFrontendClient,
...
+                def is_within_directory(app_src_path: str, target: str) -> bool:
+                    """ Check if target path of tar-files is in the application source directory. """
+                    abs_app_src_path = os.path.abspath(app_src_path)
+                    abs_target = os.path.abspath(target)
+
+                    prefix = os.path.commonprefix([abs_app_src_path, abs_target])
+
+                    return prefix == abs_app_src_path
+
+                def safe_extract(tar: Any, app_src_path: str) -> None:
```
Comment (platelminto) [inline src/adk/managers/resource_manager.py:91]:
```suggestion
                def safe_extract(tar: tarfile.TarFile, app_src_path: str) -> None:
```

File: src/adk/managers/resource_manager.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
-from pathlib import Path
 import os
+from pathlib import Path
 import tarfile
-from typing import cast, Tuple, List
+from typing import cast, Tuple, List, Any
```
Comment (platelminto) [inline src/adk/managers/resource_manager.py:4]:
```suggestion
from typing import cast, Tuple, List
```

**Review by platelminto — CHANGES_REQUESTED** [review]

File: src/adk/managers/resource_manager.py (hunk 1 above)
Comment (QFer) [reply src/adk/managers/resource_manager.py]: Nice catch

File: src/adk/managers/resource_manager.py (hunk 2 above)
Comment (QFer) [reply src/adk/managers/resource_manager.py]: Removed Any

**Review by platelminto — APPROVED** [review]

**Pull request merged**